pkg/supportbundle/supportbundle.go

```diff
@@ -129,7 +129,7 @@ func GetBundleCommand(appSlug string) []string {
 
 	command := []string{
 		"curl https://krew.sh/support-bundle | bash",
-		fmt.Sprintf("kubectl support-bundle %s --redactors=%s\n", GetSpecURI(appSlug), redactors),
+		fmt.Sprintf("kubectl support-bundle --load-cluster-specs --redactors=%s\n", redactors),
```

Comment:
I ran this command in Okteto, which is a namespace-scoped KOTS instance, and it returned the following:
```
$ kubectl support-bundle --load-cluster-specs --redactors=configmap/cbodonnell/kotsadm-redact-spec/redact-spec,configmap/cbodonnell/kotsadm-craig-helm-redact-spec/redact-spec
Error: no specs found in cluster
```

Running with `-n cbodonnell` allowed it to run.  Does this mean that the support-bundle command is namespace scoped?

Reply:
It also appears that some of the analyzers were not de-duplicated when running that way?
```
Check PASS
Title: Required Kubernetes Version
Message: Your cluster meets the recommended and required versions of Kubernetes

------------
Check PASS
Title: kotsadm Status
Message: At least 1 replica of the Admin Console API is running and ready

------------
Check PASS
Title: Required Kubernetes Version
Message: Your cluster meets the recommended and required versions of Kubernetes

------------
Check PASS
Title: kotsadm Status
Message: At least 1 replica of the Admin Console API is running and ready

------------
```

Reply:
According the code https://github.com/replicatedhq/troubleshoot/blob/546ffde14b5c7bcc883ceead3303a1f6e7dbffc2/cmd/troubleshoot/cli/run.go#L117. And a error,
I think it is using namespace "" if you don't input any namespace, but according to https://troubleshoot.sh/docs/support-bundle/discover-cluster-specs/#usage, it should search everything in the cluster. Which made me also quite confusing
```
2023/03/16 17:26:30 failed to load redactor specs from config maps: failed to search for configmaps in the cluster: configmaps is forbidden: User "system:serviceaccount:okteto:0cc534d8-64a0-4abb-b6dd-f45e055ac018" cannot list resource "configmaps" in API group "" at the cluster scope
```

Reply:
for de-duplicated, I think our deduplicatedAnalyzers https://github.com/replicatedhq/kots/blob/29c817c403807f6dd6b43ef305abdb925dd019b4/pkg/supportbundle/spec.go#L514 is not actually deduplicated. 

The right one should be like https://github.com/replicatedhq/kots/blob/1df1fb704daa0b55c809e864fe4bf432ef68b9ac/pkg/supportbundle/spec.go#L743. I will rewrite deduplicatedAnalyzers and deduplicatedCollectors according that one, if you can also confirm that our origin way is not right?

Reply:
I have tested in my ubuntu cluster with `curl https://kurl.sh/latest | sudo bash`,

It will search all the namespace.
```
 kubectl support-bundle --load-cluster-specs -n kube-flannel
 kubectl support-bundle --load-cluster-specs
```
has same result

Reply:
Error: no specs found in cluster, I think it is complain about the redactors spec cannot find

Reply:
> I will rewrite deduplicatedAnalyzers and deduplicatedCollectors according that one, if you can also confirm that our origin way is not right?

I agree, that looks like it's only checking for a handful of specific collectors/analyzers in those functions.  However, I only saw duplicated analyzers when running from the CLI with the `kubectl support-bundle` and not when using KOTS to generate the bundle.  Do you see the same behavior?

> Error: no specs found in cluster, I think it is complain about the redactors spec cannot find

Both of those redactor configmaps do exist in the namespace.  I tried adding some logging to [this section](https://github.com/replicatedhq/troubleshoot/blob/main/cmd/troubleshoot/cli/run.go#L134-L145) in troubleshoot and I see that it's not finding the support bundle specs in secrets/configmaps when I don't pass a namespace.

```
$ ./bin/support-bundle --load-cluster-specs --redactors=configmap/cbodonnell/kotsadm-redact-spec/redact-spec,configmap/cbodonnell/kotsadm-craig-helm-redact-spec/redact-spec
*** found 0 support bundle specs in secrets
*** found 0 support bundle specs in configmaps
*** found 0 support bundle specs in cluster
Error: no specs found in cluster
$ ./bin/support-bundle -n cbodonnell --load-cluster-specs --redactors=configmap/cbodonnell/kotsadm-redact-spec/redact-spec,configmap/cbodonnell/kotsadm-craig-helm-redact-spec/redact-spec
*** found 1 support bundle specs in secrets
*** found 3 support bundle specs in configmaps
*** found 4 support bundle specs in cluster
```

I don't necessarily want to hold up the review for this PR because of this.  We could change the commands back to the original and I think it would still work since we keep the old `kotsadm-<app-slug>-supportbundle` secret around for backwards compatibility.  I'll leave that up to you.

Reply:
The [helm-managed test is failing](https://github.com/replicatedhq/kots/actions/runs/4412456556/jobs/7747529870#step:8:418) is also because it's expecting the old command, so this test will need to be updated.  I can do this once we're all set here or we can pair on it if you want to learn how to do it.

Reply:
Aha! I tested ` ./bin/support-bundle --load-cluster-specs` against okteto cluster and got same error `Error: no specs found in cluster`. Then when I open the debug mode `./bin/support-bundle --load-cluster-specs --debug`, I found the issue is actually that okteto forbidden user to do a search across all the namespace, so it is not our script issue. Could you try it again with debug mode?


```
E0317 16:11:10.525947    3226 run.go:137] failed to load support bundle spec from secrets: failed to search for secrets in the cluster: secrets is forbidden: User "system:serviceaccount:okteto:0cc534d8-64a0-4abb-b6dd-f45e055ac018" cannot list resource "secrets" in API group "" at the cluster scope
E0317 16:11:10.693767    3226 run.go:143] failed to load support bundle spec from secrets: failed to search for configmaps in the cluster: configmaps is forbidden: User "system:serviceaccount:okteto:0cc534d8-64a0-4abb-b6dd-f45e055ac018" cannot list resource "configmaps" in API group "" at the cluster scope
E0317 16:11:10.862615    3226 run.go:175] failed to load redactor specs from config maps: failed to search for secrets in the cluster: secrets is forbidden: User "system:serviceaccount:okteto:0cc534d8-64a0-4abb-b6dd-f45e055ac018" cannot list resource "secrets" in API group "" at the cluster scope
E0317 16:11:11.036565    3226 run.go:181] failed to load redactor specs from config maps: failed to search for configmaps in the cluster: configmaps is forbidden: User "system:serviceaccount:okteto:0cc534d8-64a0-4abb-b6dd-f45e055ac018" cannot list resource "configmaps" in API group "" at the cluster scope

============ Collectors summary =============
Suceeded (S), eXcluded (X), Failed (F)
=============================================
No collectors executed

============ Redactors summary =============
No redactors executed

============= Analyzers summary =============
Suceeded (S), eXcluded (X), Failed (F)
=============================================
No analyzers executed

Duration: 0ms
Error: no specs found in cluster
```

Reply:
for the duplicate analyzer, I tried `kubectl support-bundle --load-cluster-specs cmd` in a purl kurl cluster. Seems it is fine. I uploaded analysis.json for your reference. OK, I will rewrite those deduplicate function
[analysis.json.txt](https://github.com/replicatedhq/kots/files/10997678/analysis.json.txt)

Reply:
For deduplicatedCollectors and deduplicatedAnalyzers, I have rewriten them with more test cases. Could you also test it again?

Reply:
Thanks @DexterYan, I'll take a look at the updated deduplicate functions.

> I found the issue is actually that okteto forbidden user to do a search across all the namespace

This is the issue that we will want to resolve.  KOTS instances can use cluster-scoped RBAC (like kURL) or they can use namespace-scoped RBAC (only applies to existing clusters - some docs [here](https://docs.replicated.com/vendor/packaging-rbac#min-rbac) and [here](https://docs.replicated.com/enterprise/installing-general-requirements#namespace-scoped)).  If an instance is namespace-scoped, we will want to provide a `support-bundle` command that will run successfully for those users. Maybe in those cases we need to add the `-n <namespace>` flag?  There is an example of how to check if KOTS is cluster or namespace-scoped [here](https://github.com/replicatedhq/kots/blob/main/pkg/supportbundle/spec.go#L308-L320).

Reply:
--load-cluster-specs has been rollbacked to use secret uri due to current issue existed in our `kubectl support-bundle --load-cluster-specs`. PRs have been created in troubleshoot to fix them. We would prefer to approach current change in kots without waiting for support-bundle changes